iam/snippets/snippets_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package snippets
+
+import (
+	"math/rand"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestViewGrantableRoles(t *testing.T) {
+	projectID := os.Getenv("GOLANG_SAMPLES_PROJECT_ID")
+	resource := "//cloudresourcemanager.googleapis.com/projects/" + projectID
+	viewGrantableRoles(resource)
```

Comment:
Check no error is returned.

Reply:
Added error checking to all calls w/ log.Fatalf, which will cause the test to fail if the call fails.

Reply:
`log.Fatal` does not work well with tests. It stops test execution completely and will not run anymore tests.

Returning errors is idiomatic Go. Calling `log.Fatal` in a library function is not.